### PR TITLE
docs(barcode):added documentation for the barcode component

### PR DESCRIPTION
**Pull request by xristianstefanov**
- Added documentation for the Barcode component

Related to: https://github.com/telerik/blazor/issues/2481

### CODE REVIEW

File: components/barcode/overview.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+The Barcode represents data in a machine-readable format. You can choose from several supported standards.
+
+All graphics are rendered on the client by using [Scalable Vector Graphics (SVG)](https://www.w3.org/Graphics/SVG/).
+
+## Basics
+
+To add a Telerik Barcode to your Blazor app:
+
+1. Add the `TelerikBarcode` tag.
+1. Inside it, add `BarcodeText` tag that styles the barcode text.
```
Comment (marin-bratanov) [inline components/barcode/overview.md:22]: Remove this point, it is not mandatory, keep the first sample simple

File: components/barcode/overview.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+1. Provide `Value` (one-way data binding) or `bind-Value` (two-way data binding) property
```
Comment (marin-bratanov) [inline components/barcode/overview.md:23]: match the example. Also, what is two-way binding for a non-editor component? If you want to talk about this here (I recommend mantioning in a later Features section), explain what that is in a dedicated paragraph or article.

File: components/barcode/encoding.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+title: Encoding
+page_title: Barcode Encoding
+description: Types of Encoding supported by the Barcode for Blazor.
+slug: components/barcode/encoding
+tags: telerik,blazor,barcode,encoding
+published: True
+position: 3
+---
+
+# Barcode Encoding
```
Comment (marin-bratanov) [inline components/barcode/encoding.md:11]: add a basic example. it is easy to put the possible options from the enum in a DDL (we have a kb and several docs articles that do this) so people can see how things work and look. At the very least, say which parametr to set for this.

File: components/barcode/overview.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+* `BarcodeText.BarcodeTextMargin.Left` - `double`.
+
+* `BarcodeText.BarcodeTextMargin.Right` - `double`.
+
+* `BarcodeText.BarcodeTextMargin.Top` - `double`.
+
+* `BarcodeText.Visible` - `bool` - by setting it to false, the Barcode will not display the value as a text below the barcode lines.
+
+### Other Properties
```
Comment (marin-bratanov) [inline components/barcode/overview.md:95]: These are more important than the appearance settings above them. Group them in further nested headings and put the appearance ones later, after the functionality ones.

File: components/barcode/overview.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+* `Class` - `string` - the CSS class that will be rendered on the main wrapping element of the Barcode component. 
+
+## Methods
+
+The Barcode methods are accessible through it's reference.
+
+* `Refresh` - redraws the component.
+
+>caption Get a reference to the Barcode and use its methods
```
Comment (marin-bratanov) [inline components/barcode/overview.md:121]: Fix the snippet to show a method usage that is real-world-like (e.g., rerendering after a resize or whatever is necessary). Remove the BarcodeText settings.

**Review by marin-bratanov — CHANGES_REQUESTED** [review]

File: components/barcode/overview.md (hunk 1 above)
Comment (xristianstefanov) [reply components/barcode/overview.md:22]: Done.

File: components/barcode/overview.md (hunk 2 above)
Comment (xristianstefanov) [reply components/barcode/overview.md:23]: I agree with you here that this is not the place for mentioning two-way binding.

File: components/barcode/overview.md (hunk 4 above)
Comment (xristianstefanov) [reply components/barcode/overview.md:95]: Done.

File: components/barcode/overview.md (hunk 5 above)
Comment (xristianstefanov) [reply components/barcode/overview.md:121]: Done.

**Review by marin-bratanov — APPROVED** [review]

**Pull request merged**